src/index.ts

```diff
@@ -154,37 +154,76 @@ export function WithFinalAuthTo<
 // @todo: implement all the other HTPipe*'s - note that each one will be slightly different based on their specifics...
 // i.e. can return bool vs not, possibly async vs sync only, mandatory vs not mandatory...
 // then the final master HTPipe will just build an object out of all the sub-HTPipe*'s
+
+type keyofTExtendsKeyofUconditionalType<T, U> = keyof T extends keyof U
```

Comment:
this is the fix for not covered right case
i.e:         const left = {
          attachData(context: { a: string }) {
            return { b: 4 };
          },
        };
        const rightNotCovered = {
          attachData(context: { other: string }) {
            return { c: 4 };
          },
        };
the problem was with out typing
i decide to avoid this problem by checking if TRight attach data params have at least one key from TLeft outputs before putting that type to TLeft HasAttachData
just in case error case with bad left working like supposed
also if we have at least one same key in left outputs and right inputs it returns Partial<Parameters<TRight['attachData']>[0]>

Reply:
I haven't completely read this commit, but I don't believe this is complete.  T and U can PARTIALLY overlap...  For example,
left: (context: {a: string, b: number}) => {c: string, d: string}
right: (context: {c: string, e: string}) => (f: string)
should become (context: {a: string, b: number, e: string}) => {d: string, f: string}